jobs/resource_job.go

```diff
@@ -249,6 +256,7 @@ type JobSettings struct {
 	PipelineTask           *PipelineTask       `json:"pipeline_task,omitempty" tf:"group:task_type"`
 	PythonWheelTask        *PythonWheelTask    `json:"python_wheel_task,omitempty" tf:"group:task_type"`
 	DbtTask                *DbtTask            `json:"dbt_task,omitempty" tf:"group:task_type"`
+	RunJobTask             *RunJobTask         `json:"run_job_task,omitempty" tf:"group:task_type"`
```

Comment:
Do we really support it in API 2.0 style?  I suggest not to do it as a top-level object and force use inside of  the `task` object

Reply:
We should support RunJobTask in API 2.0 style as customers have several use cases of running a Single Task Job as a RunJobTask. cc: @gaborratky-db

Reply:
Right now we don't have parity between supported tasks - it should be discussed maybe in the office hours

Reply:
@alexott it turns out that customers can use Run Job task for various useful things:
 * Trigger the same job with different schedules and/or parameters configured in Run Job task jobs
 * Trigger a multi-task job with dependencies continuously from a continuous Run Job task job

Most customers are also familiar with the Jobs API 2.0 syntax so let's keep this supported.